File: php/hamming/hamming.php
<?php

function distance($a, $b)
{
    $dnaStrand1 = str_split($a);
    $dnaStrand2 = str_split($b);

    if (count($dnaStrand1) !== count($dnaStrand2)) {
        throw new InvalidArgumentException('DNA strands must be of equal length.');
    }

    return count(array_diff_assoc(($dnaStrand1), ($dnaStrand2)));
}

File: ruby/hamming/hamming.rb
class Hamming
    def self.compute(strand_a, strand_b)

        if strand_a.size != strand_b.size
            raised('The strands size must be equal')
        end

        strand_a.each_char.with_index.count do |value, key|
            value != strand_b[key]
        end
        
    end

    def self.raised(message)
        raise(ArgumentError, message)
    end

end

module BookKeeping
    VERSION = 3
end